CHANGELOG.md

```diff
@@ -4,6 +4,182 @@ All notable changes to Router will be documented in this file.
+
+We have corrected these by re-visiting the way we integrate with OpenTelemetry and the supporting tracing packages and brought our usage to align with new best-practices.
+
+In addition, the testing coverage for telemetry in general has been significantly improved.  For more details of what changed and why take a look at https://github.com/apollographql/router/pull/2358.
+
+By [@bryncooke](https://github.com/bryncooke) and [@geal](https://github.com/geal) and [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2358
+
+### Metrics attributes allow value types as defined by OpenTelemetry ([Issue #2510](https://github.com/apollographql/router/issues/2510))
+
+Metrics attributes in OpenTelemetry allow the following types:
+
+* `string`
+* `string[]`
+* `float`
+* `float[]`
+* `int`
+* `int[]`
+* `bool`
+* `bool[]`
+
+However, our configuration only allowed strings. This has been fixed, and therefore it is now possible to use booleans via environment variable expansion as metrics attributes.
+
+For example:
+```yaml
+telemetry:
+  metrics:
+    prometheus:
+      enabled: true
+    common:
+      attributes:
+        supergraph:
+          static:
+            - name: "my_boolean"
+              value: ''
+```
+
+By [@bryncooke](https://github.com/bryncooke) in https://github.com/apollographql/router/pull/2616
+
+### Add missing `status` attribute on some metrics ([PR #2593](https://github.com/apollographql/router/pull/2593))
+
+When we were labeling metrics, we were not consistently adding the `status` attribute for some status codes. Instead of having an empty `status` attribute on your metrics, you'll now have `status="500"`.
+
+By [@bnjjj](https://github.com/bnjjj) in https://github.com/apollographql/router/pull/2593
+
+## 🛠 Maintenance
+
+### Upgrade to Apollo Federation v2.3.2
+
+This brings in a patch update to our Federation support, bringing it to v2.3.2.
+
+By [@abernix](https://github.com/abernix) in https://github.com/apollographql/router/pull/2586
+
+### CORS: Give a more meaningful message for users who misconfigured `allow_any_origin` ([PR #2634](https://github.com/apollographql/router/pull/2634))
+
+Allowing "any" origin in the router configuration can be done as follows:
+
+```yaml
+cors:
+  allow_any_origin: true
+```
+
+However, some intuition and familiarity with the CORS specification might also lead someone to configure it as follows:
+
+```yaml
+cors:
+  origins:
+    - "*"
+```
+
+Unfortunately, this won't work and the error message received when it was attempted was neither comprehensive nor actionable:
+
+```
+ERROR panicked at 'Wildcard origin (`*`) cannot be passed to `AllowOrigin::list`. Use `AllowOrigin::any()` instead'
+```
+
+This usability improvement adds helpful instructions to the error message, pointing you to the correct pattern for setting up this behavior in the router:
+
+```
+Invalid CORS configuration: use `allow_any_origin: true` to set `Access-Control-Allow-Origin: *`
```

Comment:
Just a comment, but I don't remember why we don't allow wildcard. If this is what people are tying to do why don't we accept it?

Reply:
Possibly a room for future improvement.  @o0Ignition0o to comment, perhaps, but probably best for a follow-up.